File: demos_python/11_color_tool/color.py
from __future__ import annotations


class Color:
    def __init__(self, name: str, hex_code: str) -> None:
        self.name = name
        self.hex_code = hex_code

    @staticmethod
    def new(name: str, hex_code: str) -> Color:
        return Color(name, hex_code)

    def print(self) -> None:
        print(f"Color: {self.name} (#{self.hex_code})")

File: demos_rust/12_color_tool_refactor/src/color.rs
pub struct Color {
    name: String,
    hex_code: String,
}

impl Color {
    pub fn new(name: &str, hex_code: &str) -> Color {
        Color {
            name: name.to_string(),
            hex_code: hex_code.to_string(),
        }
    }

    pub fn print(&self) {
        println!("Color: {} (#{})", self.name, self.hex_code);
    }
}
